types/underscore/underscore-tests.ts

```diff
@@ -821,7 +822,7 @@ declare const extractChainTypes: ChainTypeExtractor;
 
     const anyFunctionIteratee: _.Iteratee<any, string> = (element, key, collection) => {
         element; // $ExpectType any
-        key; // $ExpectType string | number
+        key; // $ExpectType any
```

Comment:
I didn't want to do this, but updating `V` in `CollectionIterator` to handle `undefined` as a possible `V` seemed like the best of the fixes that I could come up with and this was a necessary compromise since `any extends undefined` is true. Since folks should ideally be avoiding `any` anyway, I don't feel super bad about this.

An alternate solution that I considered was doing something like the below, but I feel like it's not a great solution.

```TS
interface CollectionIterator<T extends TypeOfCollection<V, any>, TResult, V = Collection<T>, TOptionalCollection extends never | undefined = never> {
    (element: T, key: CollectionKey<V> | TOptionalCollection, collection: V | TOptionalCollection): TResult;
}
```

Reply:
I agree `never | undefined` doesn't look great.